[backport] Integrate benchmark suite into main SBT build.

retronym opened the pull request:
This makes life easier when switching branches, and catches bitrot in the benchmark suite by compiling it in CI.

retronym committed:
Improvements to the IntellIJ config.

  - Align the junit/test output path with SBT's. This allows you to
    compile in SBT and trigger a test execution (including debugging)
    in IntelliJ with a run configuration that has "build" removed as
    a pre-run ste.
  - Add the benchmarks subproject. This gives autocomplete and build
    within IntelliJ. Benchmarks still need to be run under SBT, however,
    to let sbt-jmh do the code generation.
  - Add some more auto-imports to our hand-rolled definitions of the
    scala-build project, to eliminate some highlighting errors in
    build.sbt

(cherry picked from commit cac5a86bae0e05b7e080aa6a78f97ba5351096ff)

retronym committed:
Integrate benchmarks into the main build

and compile the benchmarks in testAll

(cherry picked from commit 7b85527ed578bb9a0670af2e0621cb4ec2c325b3)

dwijnand commented on build.sbt:
```diff
@@ -238,8 +242,7 @@ lazy val commonSettings = clearSourceAndResourceDirectories ++ publishSettings +
 
   // Don't log process output (e.g. of forked `compiler/runMain ...Main`), just pass it
   // directly to stdout
-  outputStrategy in run := Some(StdoutOutput),
```
Traily McComma can stay, ya know 😄

dwijnand commented:
Backport of #6528 (as a starting point)

retronym committed:
[backport] +compiler/reflect the the bench classpath

Partial backport of #6622

Restores a ProdConsBenchmark now that it will compile.
